### PR TITLE
Create build-on-pull-request.yml

### PR DESCRIPTION
**Pull request by csc-gip**

### CODE REVIEW

File: .github/workflows/build-on-pull-request.yml [hunk 1]
```diff
@@ -21,30 +14,28 @@ jobs:
...
-          cd installation
-          source ./build.sh 2>&1 > /dev/null
-          prepare_modules
           
      - name: determine changed modules
        env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
-          gh pr diff ${{ inputs.pr-number }}${{ github.event.pull_request.number }} --name-only | grep '^modules/' | cut -f 2-3 -d/ | sort | uniq | tee changed-modules.txt
+          gh pr diff {{ github.event.pull_request.number }} --name-only | grep '^modules/' | cut -f 2-3 -d/ | sort | uniq | tee changed-modules.txt
```
Comment (csc-gip) [inline .github/workflows/build-on-pull-request.yml:29]: ```$``` is missing

File: .github/workflows/build-on-pull-request.yml [hunk 2]
```diff
@@ -21,30 +14,28 @@ jobs:
          java-version: 11
          distribution: 'zulu'
 
+     - name: Use Node.js 18.10
```
Comment (csc-gip) [inline .github/workflows/build-on-pull-request.yml:17]: We do not build the frontend and do not need node

File: .github/workflows/build-on-pull-request.yml [hunk 3]
```diff
@@ -21,30 +14,28 @@ jobs:
...
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
-          gh pr diff ${{ inputs.pr-number }}${{ github.event.pull_request.number }} --name-only | grep '^modules/' | cut -f 2-3 -d/ | sort | uniq | tee changed-modules.txt
+          gh pr diff {{ github.event.pull_request.number }} --name-only | grep '^modules/' | cut -f 2-3 -d/ | sort | uniq | tee changed-modules.txt
+
+     - name: build basic factory
+       run: |
+          cd installation
+          ./build.sh build
+       if: $(cat changed-modules.txt | wc -l) != 0
```
Comment (csc-gip) [inline .github/workflows/build-on-pull-request.yml]: I'd like to at least build the factory as a check on every pr

File: installation/build.sh [hunk 4]
```diff
@@ -631,6 +630,7 @@ build() {
   build_persistencelayers
   build_oracle_aq_tools
   fill_lib
+  prepare_modules
```
Comment (csc-gip) [inline installation/build.sh:633]: I think this will run it twice, when all is built

File: installation/build.sh [hunk 5]
```diff
@@ -28,9 +29,8 @@ check_dependencies() {
   mvn --version
   ant -version
   git --version
+  node --version
```
Comment (csc-gip) [inline installation/build.sh:32]: npm --version?
We should check npm only if we build the frontend

**Review by LukasFey-GIP — APPROVED** [review]

**Pull request merged**